File: app/models/enrolment/ExternalReservation.java
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package models.enrolment;

import controllers.facility.RoomLike;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.OneToOne;
import models.base.GeneratedIdentityModel;
import models.facility.MailAddress;

@Entity
public class ExternalReservation extends GeneratedIdentityModel implements RoomLike {

    private String orgRef;

    private String orgName;

    private String orgCode;

    private String roomRef;

    @OneToOne(cascade = CascadeType.ALL)
    private MailAddress mailAddress;

    private String buildingName;

    private String campus;

    private String machineName;

    private String roomName;

    private String roomCode;

    private String roomTz;

    private String roomInstruction;

    private String roomInstructionEN;

    private String roomInstructionSV;

    public String getOrgRef() {
        return orgRef;
    }

    public void setOrgRef(String orgRef) {
        this.orgRef = orgRef;
    }

    public String getOrgName() {
        return orgName;
    }

    public void setOrgName(String orgName) {
        this.orgName = orgName;
    }

    public String getOrgCode() {
        return orgCode;
    }

    public void setOrgCode(String orgCode) {
        this.orgCode = orgCode;
    }

    public String getRoomRef() {
        return roomRef;
    }

    public void setRoomRef(String roomRef) {
        this.roomRef = roomRef;
    }

    public String getMachineName() {
        return machineName;
    }

    public void setMachineName(String machineName) {
        this.machineName = machineName;
    }

    public String getRoomName() {
        return roomName;
    }

    public void setRoomName(String roomName) {
        this.roomName = roomName;
    }

    public String getRoomCode() {
        return roomCode;
    }

    public void setRoomCode(String roomCode) {
        this.roomCode = roomCode;
    }

    public String getRoomTz() {
        return roomTz;
    }

    public void setRoomTz(String roomTz) {
        this.roomTz = roomTz;
    }

    public MailAddress getMailAddress() {
        return mailAddress;
    }

    public void setMailAddress(MailAddress mailAddress) {
        this.mailAddress = mailAddress;
    }

    public String getBuildingName() {
        return buildingName;
    }

    public void setBuildingName(String buildingName) {
        this.buildingName = buildingName;
    }

    public String getCampus() {
        return campus;
    }

    public void setCampus(String campus) {
        this.campus = campus;
    }

    @Override
    public String getRoomInstruction() {
        return roomInstruction;
    }

    public void setRoomInstruction(String roomInstruction) {
        this.roomInstruction = roomInstruction;
    }

    @Override
    public String getRoomInstructionEN() {
        return roomInstructionEN;
    }

    public void setRoomInstructionEN(String roomInstructionEN) {
        this.roomInstructionEN = roomInstructionEN;
    }

    @Override
    public String getRoomInstructionSV() {
        return roomInstructionSV;
    }

    public void setRoomInstructionSV(String roomInstructionSV) {
        this.roomInstructionSV = roomInstructionSV;
    }
}

File: app/models/enrolment/ExternalReservation.scala
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package models.enrolment

import jakarta.persistence.{CascadeType, Entity, OneToOne}
import models.base.GeneratedIdentityModel
import models.facility.{MailAddress, RoomLike}

import scala.compiletime.uninitialized

@Entity
class ExternalReservation extends GeneratedIdentityModel with RoomLike:
  var orgRef: String  = uninitialized
  var orgName: String = uninitialized
  var orgCode: String = uninitialized
  var roomRef: String = uninitialized

  @OneToOne(cascade = Array(CascadeType.ALL))
  var mailAddress: MailAddress = uninitialized

  var buildingName: String      = uninitialized
  var campus: String            = uninitialized
  var machineName: String       = uninitialized
  var roomName: String          = uninitialized
  var roomCode: String          = uninitialized
  var roomTz: String            = uninitialized
  var roomInstruction: String   = uninitialized
  var roomInstructionEN: String = uninitialized
  var roomInstructionSV: String = uninitialized
